store/src/db/mod.rs

```diff
@@ -130,4 +131,36 @@ impl Db {
             .await
             .map_err(|_| anyhow!("Get account hashes task failed with a panic"))?
     }
+
+    pub async fn apply_block(
+        &self,
+        allow_acquire: oneshot::Sender<()>,
+        acquire_done: oneshot::Receiver<()>,
```

Comment:
i looked at multiple alternatives. the best approach here would be to use a monitor / condvar, but for the libraries that I look there was no implementation that worked with async code (and here we need something that works with async *and* sync, since rusqlite runs in a sync thread)